### PR TITLE
Support python>3.8 for verp emails and reduce size by truncating hmac and storing time in minutes since 2022-01-01

### PR DESCRIPTION
**Pull request by acasajus**
- For python > 3.9 `shake128` has a new required parameter when generating the digest that breaks the `hmac` module. So we use `sha3-224` and truncate the resulting hmac to 8 bytes.
- Store time in minutes from 2022-01-01 to reduce further the length of the generated email.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        python-version: [3.7]
+        python-version: [3.7, 3.9]
```
Comment (nguyenkims) [inline .github/workflows/main.yml:11]: 👌

**Comment by nguyenkims** [comment]
@acasajus does this mean we change the algorithm to generate envelope mail_from? Can that affect how we parse the previously generated mail_from? In this case maybe we need to handle the previous format as well? (the same way we did for handling the old `bounce+{email_log_id}@sl.co`)

**Comment by acasajus** [comment]
Has the old one been in production already?

**Comment by nguyenkims** [comment]
@acasajus yep it's been deployed yesterday.

**Comment by acasajus** [comment]
Argh.. ok. I'll add backwards compat

File: app/email_utils.py [hunk 2]
```diff
@@ -1513,3 +1521,38 @@ def get_verp_info_from_email(email: str) -> Optional[Tuple[VerpType, int]]:
     if data[2] > time.time() + VERP_MESSAGE_LIFETIME:
         return None
     return VerpType(data[0]), data[1]
+
+
+# This method processes the email address, checks if it's a signed verp email generated by us to receive bounces
```
Comment (nguyenkims) [inline app/email_utils.py:1526]: Nits: can we move these comments to the method doc?

**Review by nguyenkims — APPROVED** [review]
LGTM! Just left a suggestion. Is the algo change validated by crypto team?

**Comment by acasajus** [comment]
> LGTM! Just left a suggestion. Is the algo change validated by crypto team?

yes :)

File: app/email_utils.py (hunk 2 above)
Comment (acasajus) [reply app/email_utils.py]: Done

**Pull request merged**